contrib/src/main/scala/akka/stream/contrib/SourceGen.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.stream._
+import akka.stream.scaladsl._
+import akka.stream.stage._
+
+/**
+ * Source factory methods are placed here
+ */
+object SourceGen {
+
+  /**
+   * Create a `Source` that will unfold a value of type `S` by
+   * passing it through a flow. The flow should emit a
+   * pair of the next state `S` and output elements of type `E`.
+   * Source completes when the flow completes.
+   */
+  def unfoldFlow[S, E, M](seed: S)(flow: Graph[FlowShape[S, (S, E)], M]): Source[E, M] = {
```

Comment:
Doesn't it make sense to implement this in terms of the other one with the Option? I know it adds some overhead, but there are allocations because of a Pair anyway and I don't think an extra Map is that costly. Or, at least benchmark the two options whether it is worth to keep them quasi-separate implementations.